modules/finality-verifier/src/lib.rs

```diff
@@ -119,18 +122,27 @@ pub mod pallet {
 				voter_set,
 				&justification,
 			)
-			.map_err(|_| <Error<T>>::InvalidJustification)?;
+			.map_err(|e| {
+				frame_support::debug::error!("Received invalid justification for {:?}: {:?}", finality_target, e);
+				<Error<T>>::InvalidJustification
+			})?;
 
 			let best_finalized = T::HeaderChain::best_finalized();
+			frame_support::debug::trace!("Checking ancestry against best finalized header: {:?}", &best_finalized);
+
 			ensure!(
 				T::AncestryChecker::are_ancestors(&best_finalized, &finality_target, &ancestry_proof),
 				<Error<T>>::InvalidAncestryProof
 			);
 
 			// Note that this won't work if we ever change the `ancestry_proof` format to be
 			// sparse since this expects a contiguous set of finalized headers.
-			let _ =
-				T::HeaderChain::append_finalized_chain(ancestry_proof).map_err(|_| <Error<T>>::FailedToWriteHeader)?;
+			let _ = T::HeaderChain::append_finalized_chain(ancestry_proof).map_err(|_| {
+				frame_support::debug::error!("Failed to append finalized header chain.");
+				<Error<T>>::FailedToWriteHeader
+			})?;
+
+			frame_support::debug::info!("Succesfully imported finalized header chain!");
```

Comment:
Perhaps add some indication what got actually imported, like the `finality_target`.